### PR TITLE
csv-invert: Perf improvments!

### PR DESCRIPTION
**Pull request by cmasone-attic**
Tweaking the main loop that processes list entries to avoid some
map assignments, lookups, and allocations saves 15% or so, resulting
in an overall savings of about 1m on the 6m runtime of our test
workload (as run on my laptop).

Towards #3690

### CODE REVIEW

File: go/types/sequence_chunker.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (sc *sequenceChunker) createSequence() (sequence, metaTuple) {
 	seq := col.sequence()
 	mt := newMetaTuple(sc.vrw.WriteValue(col), key, numLeaves)
 
-	sc.current = []sequenceItem{}
+	sc.current = sc.current[:0]
```
Comment (cmasone-attic) [inline go/types/sequence_chunker.go]: Since this is a more general-purpose perf win, I wonder if it should be in a separate PR? @rafael-atticlabs, thoughts?

**Comment by cmasone-attic** [comment]
PTAL

File: go/types/sequence_chunker.go (hunk 1 above)
Comment (rafael-atticlabs) [reply go/types/sequence_chunker.go]: yeah, kalman has a patch outstanding in this area. I'll ask him to do this as a part of that.

Comment (cmasone-attic) [reply go/types/sequence_chunker.go]: ok, removing

**Comment by cmasone-attic** [comment]
well, the win will be less, now, that the sequence_chunker.go change is put off to another patch. But, still we should get about a 15% improvement
